dls-controls/malcolmjs
# Feature/method param behaviour #352

## Description

Improved behaviour of method input parameters:
- parameters which are to be sent are marked as dirty
- required parameters are preset to their specified or meta type default value
- non-required parameters are set to their default value but are not sent unless modified from this
- method buttons are marked as dirty if any of the input parameter values are different from the last submitted values
- added a "discard params" button to the overall and individual parameter method info panes

## Testing instructions

Add a set up instructions describing how the reviewer should test the code
- Review code
- Check Travis build
- Review changes to test coverage
- npm run ui-only
- connect to a malcolm instance which has some methods
- check that any required params are intially flagged as dirty
- try editing a method input parameter and check that both the individual parameter and method run button are marked as dirty
- submit the method and check that the dirty icon on the button goes away when a successful return is received 
- open the info pane for the method and click the "discard params" button under the input parameters expander
- check that all the individual parameters are no longer flagged as dirty

## Agile board tracking

connect to #352 
closes #352
